src/ImapLibrary2/__init__.py

```diff
@@ -461,6 +468,14 @@ def _criteria(**kwargs):
             criteria += ['CC', '"%s"' % cc]
         if text:
             criteria += ['TEXT', '"%s"' % text]
+        if since:
+            since_date = datetime.strptime(since, date_format)
+            since_date = format_datetime(since_date, "d-MMM-yyyy", locale='en') # Convert to English format for IMAP 
```

Comment:
Why is this format_datetime from Babel needed if datetime module already uses the correct format? What is the difference?

Reply:
It's needed if your Python locale is not 'en'. IMAP only supports the English abbreviation for the month. In my case the Python locale is Dutch.

Reply:
Ok, seems like givin locale to datetime module is not possible https://stackoverflow.com/a/24070673
What a weird design.

So I would like to avoid any extra dependencies which Babel is, but I wonder if using that Locale-setting example correctly is too complex when compared to just using this Babel method.

Reply:
Maybe you could first save locale then set it to 'en' use it and set it back?

Reply:
I think that would be enough here since I don't think this library needs multi-threading support.

Reply:
Pabot uses different processes as I understood it, so it should be safe.